project/apps/web/stores/useSocketStore.ts

```diff
@@ -0,0 +1,64 @@
+import { io, Socket } from "socket.io-client";
+import { create } from "zustand";
+
+const SOCKET_SERVER_URL =
+  process.env.NEXT_PUBLIC_MATCH_SOCKET_URL || "http://localhost:8080";
+
+interface SocketState {
+  socket: Socket | null;
+  isConnected: boolean;
+  connectionError: string | null;
+  connect: () => void;
+  disconnect: () => void;
+}
+
+const useSocketStore = create<SocketState>((set, get) => ({
```

Comment:
er for some reason Socket type doesnt work with the `createSelectors`, actually i realised instead of needing to use selectors we can just do something like this `const { connect, disconnect } = useSocketStore();`, i think it might be easier LOL